### PR TITLE
Fix user capabilities check for the Site Editor

### LINKED ISSUE #61419: Site editor related links are exposed to non-admin users
### Description

When generating a link to a page in the site editor, it is determined whether the site editor can be accessed, mainly through the following code, and the link destination is changed depending on the result.

https://github.com/WordPress/gutenberg/blob/771acf45c3ce5b76ffe529f8a0467d1fdc1c9359/packages/edit-post/src/components/header/more-menu/manage-patterns-menu-item.js#L11-L23

However, in the latest version of Gutenberg, `canUser( 'read', 'templates' )` returns `true` for non-administrator users, and they are now considered to have "access to the site editor." As a result, when you run the command, you will receive an error indicating that access is not possible.

### Step-by-step reproduction instructions

- Create a user with the Editor role and log in as that user.
- Open the post editor.
- When you start the command, you should see items like Styles, Templates, etc.
- If you click on those items, you should see an error.
- The "Manage all patterns" link should give you a similar error.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/54422211/6a8088ab-aa3a-46b2-83d2-8053324913b3

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #61419

## What?

This PR correctly checks for non-admin user capabilities for commands and link buttons related to the site editor. This fixes an issue where unintended commands and links are displayed for non-admin users.

## Why?

Previously, determining whether the site editor was accessible depended on having permission to "read" templates. However, my understanding is that in #60326, permission to read templates was added for users with `edit_post` permissions.

This exposes commands and links that users cannot access even though they do not have `edit_theme_options` permission.

## How?

Related commands and links now check for "create" permission instead of "read" permission. This should now correctly determine whether users have the `edit_theme_options` permission as before.

## Testing Instructions

Open the post editor as an admin user and as an editor user and test the following.

#### Command

- Patterns: Links to the Site Editor's Patterns page.
- Navigation, Styles, Pages, Templates: Displayed as before.

#### Link button

All links below link to the Site Editor's Patterns page.

- Options > Manage patterns
- Create a synced pattern > Block Toolbar > Manage patterns

#### Command

- Patterns: Link to `edit.php?post_type=wp_block`.
- Navigation, Styles, Pagesm Templates: Not displayed.

#### Link button

All links below link to `edit.php?post_type=wp_block`.

- Options > Manage patterns
- Create a synced pattern > Block Toolbar > Manage patterns

> [!NOTE]
> While creating this PR, I also noticed that individual pages, individual templates, and individual template parts were exposed to non-admin users via the command palette. These are caused by the fact that there is no permission check in the first place, so I would like to fix it in a follow-up PR.

permission-check

### CODE REVIEW

**Review by fabiankaegy — APPROVED** [review]
Looks good to me 👍

**Review by Mamaduka — APPROVED** [review]

File: packages/core-commands/src/hooks.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { useSelect } from '@wordpress/data';
 
 export function useIsTemplatesAccessible() {
 	return useSelect(
-		( select ) => select( coreStore ).canUser( 'read', 'templates' ),
+		( select ) => select( coreStore ).canUser( 'create', 'templates' ),
```
Comment (youknowriad) [inline packages/core-commands/src/hooks.js]: The name of the hook is not congruent with what it does. I understand that it's an internal hook so it's not that important but maybe the hook should be renamed?

Comment (t-hamano) [reply packages/core-commands/src/hooks.js]: Indeed. For example, is `useCanCreateTemplates` appropriate?

Comment (youknowriad) [reply packages/core-commands/src/hooks.js]: Works for me or even remove the hook entirely. Not sure how useful is it to abstract it.

Comment (t-hamano) [reply packages/core-commands/src/hooks.js]: > remove the hook entirely

That makes sense. I would like to merge this PR after removing this hook and switching to an ad hoc approach.

**Pull request merged**